homeassistant/components/wallbox/coordinator.py

```diff
@@ -101,10 +119,10 @@ async def async_validate_input(self) -> None:
         """Get new sensor data for Wallbox component."""
         await self.hass.async_add_executor_job(self._validate)
 
+    @_require_authentication
     def _get_data(self) -> dict[str, Any]:
         """Get new sensor data for Wallbox component."""
         try:
-            self._authenticate()
```

Comment:
Previously, the `ConnectionError` was converted into a `UpdateFailed`. Is this still required?

Reply:
I believe so, If you can login, but updating fails -> UpdateFailed should be raised. These are seperate API endpoints, this is a possibility.

Reply:
Then you should not use the decorator here or extend the decorator to pass a custom exception

Reply:
hmm, gave this some thought. Not using the decorator here, means adding the self.authenticate() here, which means reintroducing the try...except there -> not very nice. I think its thus better to remove the UpdateFailed here -> I always understood that UpdateFailed starts a retry for the update. However, the data is updated every 30 seconds anyway, so, maybe not that useful.